library/core/src/str/traits.rs

```diff
@@ -210,9 +211,10 @@ unsafe impl SliceIndex<str> for ops::Range<usize> {
 
         // SAFETY: the caller guarantees that `self` is in bounds of `slice`
         // which satisfies all the conditions for `add`.
-        let ptr = unsafe { slice.as_ptr().add(self.start) };
-        let len = self.end - self.start;
-        ptr::slice_from_raw_parts(ptr, len) as *const str
+        unsafe {
+            let new_len = unchecked_sub(self.end, self.start);
```

Comment:
Does this get different codegen if you just use the public stdlib [`unchecked_sub`](https://doc.rust-lang.org/std/primitive.usize.html#method.unchecked_sub)?

Reply:
Ah okay! Cool then.

Reply:
No, but it doubles up on the precondition check.